src/qoi/src/rayfire_mesh.C

```diff
@@ -103,15 +101,27 @@ namespace GRINS
         next_elem = this->get_next_elem(prev_elem,start_point,end_point);
 
         // add end point as node on the rayfire mesh
-        _mesh->add_point(end_point,node_id);
+        end_node = _mesh->add_point(end_point);
         libMesh::Elem* elem = _mesh->add_elem(new libMesh::Edge2);
-        elem->set_node(0) = _mesh->node_ptr(node_id-1);
-        elem->set_node(1) = _mesh->node_ptr(node_id++);
+        elem->set_node(0) = start_node;
+        elem->set_node(1) = end_node;
+        
+        // warn if rayfire elem is shorter than TOLERANCE
+        if ( (start_point-end_point).norm() < libMesh::TOLERANCE)
+          {
+            libmesh_do_once(
+              std::cout <<"********\n"
```

Comment:
Instead of using `libmesh_do_once()`, you can use `libmesh_warning()` which can take a string for your message, but will also print out line, file, etc. and internally uses `libmesh_do_once`.